### PR TITLE
Fixes #3642: sdInit() now called in one place only (inside the opentx…

### LINKED ISSUE #3642: Model audio files not played
Model audio files (for example flight mode on/off) are not played when the radio is powered up. If the model is reloaded, then the audio files start playing. The problem is only on the radio (X9E), `simu` works just fine.

This was not a problem on 2.1.

### PR DESCRIPTION
**Pull request by projectkk2glider**
…Init()). Some reordering of opentxInit().

Please review carefully and test. I only tested on Taranis. 

The SD card is now initialized in `opentxInit()`. Before this change `referenceModelAudioFiles()` was being executed before SD was initialized. 

I also rearanged the flow of   `opentxInit()` to make it more simple and clear. 

I think  there was also bug - the `topbar = new Topbar(&g_model.topbarData)` was callled before the model data was read.

### CODE REVIEW

File: radio/src/opentx.cpp [hunk 1]
```diff
@@ -2488,16 +2488,14 @@ void opentxInit(OPENTX_INIT_ARGS)
...
+    sdInit();
 #endif
-
-#if defined(COLORLCD)
-  topbar = new Topbar(&g_model.topbarData);
-  luaInit();
-#endif
+  }
 
 #if defined(RAMBACKUP)
```
Comment (bsongis) [inline radio/src/opentx.cpp]: I don't remember what happens if lua is not initialized here, in case the screens have lua widgets inside

**Comment by bsongis** [comment]
I like this PR, just need to check or test for the remark I wrote above

**Comment by projectkk2glider** [comment]
I don't like the fact that `luaInit()` is called from two different places (depending on the radio). I will look into it to see if some code simplification and unification can be done.

File: radio/src/gui/horus/topbar.cpp [hunk 2]
```diff
@@ -20,6 +20,9 @@
 
 #include "opentx.h"
 
+
+static Topbar * topbar;
```
Comment (bsongis) [inline radio/src/gui/horus/topbar.cpp]: = 0 is not mandatory but good practice

File: radio/src/gui/horus/topbar.cpp [hunk 3]
```diff
@@ -88,6 +91,9 @@ void drawTopBar()
     lcdDrawSolidFilledRect(LCD_W-122+4*i, 30, 2, 8, i >= bars ? MENU_TITLE_DISABLE_COLOR : MENU_TITLE_COLOR);
   }
 
+  if (!topbar) {
```
Comment (bsongis) [inline radio/src/gui/horus/topbar.cpp]: Why did you move it there?

Comment (projectkk2glider) [inline radio/src/gui/horus/topbar.cpp]: It is the only place where variable `topbar` is used.

Comment (bsongis) [inline radio/src/gui/horus/topbar.cpp]: I think I would prefer a ```topbarInit()``` function, no?

**Comment by projectkk2glider** [comment]
I am not sure I like what I did with the topbar. 

Lua will need more elaborate checking/refactoring. It is a bit of a mess right now with the Hours working differently than Taranis.  There also looks like permanent scripts are not loaded on Horus.

File: radio/src/opentx.cpp [hunk 4]
```diff
@@ -2552,13 +2548,7 @@ void opentxInit(OPENTX_INIT_ARGS)
 
   if (g_eeGeneral.backlightMode != e_backlight_mode_off) backlightOn(); // on Tx start turn the light on
 
-  if (UNEXPECTED_SHUTDOWN()) {
-#if !defined(CPUARM)
-    // is done above on ARM
```
Comment (bsongis) [inline radio/src/opentx.cpp]: FYI it was done this way (yes, difficult to understand) to save some bytes of flash on 9X. I agree on revoving this

**Comment by bsongis** [comment]
Integrated into #3674
